Bio/Seq.py

```diff
@@ -3136,15 +3137,18 @@ def reverse_complement(sequence):
     >>> reverse_complement("A")
     'T'
     """
-    return complement(sequence)[::-1]
+    try:
+        return sequence.reverse_complement()  # Seq, MutableSeq, or SeqRecord
```

Comment:
So what happens if the input is a MutableSeq? Doesn't this change bypass the special case handling in the complement function.

(It this passed the tests, then I think we're missing something in our coverage - or I am not looking at this closely enough)

Reply:
Good point; `reverse_complement` indeed acts in-place for `MutableSeq`.

We should clean up the different versions of `reverse_complement`. Now, it works differently on `Seq` and `MutableSeq`, and does not accept mixed DNA/RNA sequences, while `reverse_complement_rna` does.

Reply:
And our tests did not pick up this bug.

Reply:
@peterjc 

Do you think we should modify `reverse_complement` with the appropriate warnings, or define new methods & function `reverse_complement_dna`?

Reply:
Adding an explicit ``..._dna`` set of methods/functions might be nice, but we have so much historical usage I would want to keep ``reverse_complement`` alive and tweak it as necessary.

Reply:
OK, I'll try that in a separate PR. Closing this one.